lib/reverse_sentence.rb

```diff
@@ -1,6 +1,31 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2)
+# Space complexity: O(1)
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return nil if my_sentence == nil
+  cloned_sentence = my_sentence.clone
+  index_from_begining = 0
+  word = ""
+  cloned_sentence.length.times do |i|
+    index = -1 - i
+    if cloned_sentence[index] == " "
+      my_sentence[index_from_begining] = cloned_sentence[index]
+      index_from_begining += 1
+    elsif cloned_sentence[index - 1] != " "
+      word.insert(0,cloned_sentence[index])
```

Comment:
Note that inserting characters at the beginning of an array or string is an O(n) operation as all subsequent elements have to shift down one element.